### DISCUSSION TITLE
How to change the default GameBoy resolution?

### DISCUSSION THREAD

**Discussion by ZihangYang** [deltabeard/Peanut-GB#114] [Q&A] [answered]
Hi! Thanks for writing this open source emulator. I am using the same LCD module (base on ili9225 driver). But I want to know if it is possible to increase the screen area. 

The 2-inch LCD screen is 220*176, but GameBoy emulator default resolution is 160*144. Therefore, there is empty space on the edge of the screen. I wonder if it is supported to adjust the resolution in the emulator? If so, I can probably consider to swap a bigger screen and write a new driver.

Thank you again!

**Comment by ZihangYang** [comment]
1 more thing, I found that it can't save the progress after power off. For example, if I quit the emulator, and reload the game (pokemon red), it still saves my progess. But after power off/on, all progress will be lost. Is there a way to save game progress somewhere? maybe external memory, like SD card

**Reply by deltabeard** [reply]
> But after power off/on, all progress will be lost.

That's unlikely an issue with Peanut-GB but with the front-end application that's using it. Where is your code? My use of Peanut-GB on the RP2040 has it upload the save data to an FRAM chip every second using DMA and it worked great.

**Reply by ZihangYang** [reply]
Thanks for the reply! I solved this issue

**Comment by deltabeard** [comment, marked as answer]
Peanut-GB renders the display at the exact resolution of the Game Boy. You need to use some scaling algorithm, like bilinear or nearest-neighbour to fill your LCD screen. Peanut-GB does not have a scaling feature embedded within it to do this for you.
